airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -0,0 +1,420 @@
+        :type nodegroupName: str
+        :param subnets: The subnets to use for the Auto Scaling group that is created for your nodegroup.
+        :type subnets: List[str]
+        :param nodeRole: The Amazon Resource Name (ARN) of the IAM role to associate with your nodegroup.
+        :type nodeRole: str
+
+        :return: Returns descriptive information about the created EKS Managed Nodegroup.
+        :rtype: Dict
+        """
+        eks_client = self.conn
+        # The below tag is mandatory and must have a value of either 'owned' or 'shared'
+        # A value of 'owned' denotes that the subnets are exclusive to the nodegroup.
+        # The 'shared' value allows more than one resource to use the subnet.
+        tags = {'kubernetes.io/cluster/' + clusterName: 'owned'}
+        if "tags" in kwargs:
+            tags = {**tags, **kwargs["tags"]}
+            kwargs.pop("tags")
+
+        response = eks_client.create_nodegroup(
+            clusterName=clusterName,
+            nodegroupName=nodegroupName,
+            subnets=subnets,
+            nodeRole=nodeRole,
+            tags=tags,
+            **kwargs,
+        )
+
+        self.log.info(
+            "Created a managed nodegroup named %s in cluster %s",
+            response.get('nodegroup').get('nodegroupName'),
+            response.get('nodegroup').get('clusterName'),
+        )
+        return response
+
+    def delete_cluster(self, name: str) -> Dict:
+        """
+        Deletes the Amazon EKS Cluster control plane.
+
+        :param name: The name of the cluster to delete.
+        :type name: str
+
+        :return: Returns descriptive information about the deleted EKS Cluster.
+        :rtype: Dict
+        """
+        eks_client = self.conn
+
+        response = eks_client.delete_cluster(name=name)
+
+        self.log.info("Deleted cluster with the name %s.", response.get('cluster').get('name'))
+        return response
+
+    def delete_nodegroup(self, clusterName: str, nodegroupName: str) -> Dict:
+        """
+        Deletes an Amazon EKS Nodegroup from a specified cluster.
+
+        :param clusterName: The name of the Amazon EKS Cluster that is associated with your nodegroup.
+        :type clusterName: str
+        :param nodegroupName: The name of the nodegroup to delete.
+        :type nodegroupName: str
+
+        :return: Returns descriptive information about the deleted EKS Managed Nodegroup.
+        :rtype: Dict
+        """
+        eks_client = self.conn
+
+        response = eks_client.delete_nodegroup(clusterName=clusterName, nodegroupName=nodegroupName)
+
+        self.log.info(
+            "Deleted nodegroup named %s from cluster %s.",
+            response.get('nodegroup').get('nodegroupName'),
+            response.get('nodegroup').get('clusterName'),
+        )
+        return response
+
+    def describe_cluster(self, name: str, verbose: Optional[bool] = False) -> Dict:
```

Comment:
I'm very new to type hints and wonder if it's more common to use `Optional` than this:
```suggestion
    def describe_cluster(self, name: str, verbose: bool = False) -> Dict:
```

May I have your advice on this, please? Thanks in advance! :)

Reply:
> You can use the Optional type modifier to define a type variant that allows None, such as Optional[int] (Optional[X] is the preferred shorthand for Union[X, None]):

https://mypy.readthedocs.io/en/latest/kinds_of_types.html#optional-types-and-the-none-type

I agree. We should use `bool` type here.

Reply:
Sorry for the delay, I'll make the change.

Reply:
Found a few other places where I think I misused Optional, I'll be dropping a commit in here with those corrections later today or tomorrow.

Reply:
@zkan - should be addressed in https://github.com/apache/airflow/pull/16571/commits/d63063932c4e42432a039fc69d9c592ce08b95c6